src/main/kotlin/domain/RaceJudge.kt

```diff
@@ -0,0 +1,27 @@
+package domain
+
+class RaceJudge {
+    fun findWinners(cars: Cars): Winners {
+        if (cars.getCars().isEmpty()) {
+            return Winners(emptyList())
+        }
+
+        val maxDistance = calculateMaxDistance(cars)
+        return filterWinners(cars, maxDistance)
+    }
+
+    private fun calculateMaxDistance(cars: Cars): Int {
+        return cars.getCars()
+            .maxOf { it.currentDistance() }
+    }
+
+    private fun filterWinners(
+        cars: Cars,
+        maxDistance: Int,
+    ): Winners {
+        val winners =
+            cars.getCars()
+                .filter { it.currentDistance() == maxDistance }
+        return Winners(winners)
+    }
+}
```

Comment:
아래와 같이 Kotlin Collection API를 활용하면 함수 체이닝을 통해 개발자가 읽어야 하는 코드 양을 대폭 줄일 수 있습니다. 
제안일 뿐이니 반영 여부는 규원님이 선택해주세요!

```suggestion
class RaceJudge {
    fun findWinners(cars: Cars): Winners {
        val maxDistance = cars
            .getCars()
            .maxOfOrNull { it.currentDistance() } ?: return Winners(emptyList())
        val winners = cars
            .getCars()
            .filter { it.currentDistance() == maxDistance }
        return Winners(winners)
    }
}
```

Reply:
👍 메소드를 추출해서 가독성을 올리려고 했었는데, 지금같은 경우는 처음 코드를 읽는 사람에게는 오히려 읽어야 하는 코드양이 증가하고 함수를 추적해 그 역할을 파악해야 하며, 다시 원래 함수로 돌아와야 해서 코드를 읽는 피로도가 더 증가하겠네요. 
합치는게 나아보입니다 👍